web-api/src/persistence/dynamo/users/getAccountConfirmationCode.ts

```diff
@@ -0,0 +1,22 @@
+import { AccountConfirmationRecord } from '@web-api/persistence/dynamo/dynamoTypes';
+import { ServerApplicationContext } from '@web-api/applicationContext';
+import { get } from '../../dynamodbClientService';
+
+export const getAccountConfirmationCode = async (
+  applicationContext: ServerApplicationContext,
+  {
+    userId,
+  }: {
+    userId: string;
+  },
+): Promise<string | undefined> => {
+  const result: AccountConfirmationRecord = await get({
+    Key: {
+      pk: `user|${userId}`,
+      sk: 'account-confirmation-code',
+    },
+    applicationContext,
+  });
+
+  return result?.confirmationCode;
```

Comment:
DynamoDB doesn't clean up items with a ttl precisely as they expire, and so it may be best to check to see if `result.ttl < now` value before returning the code.

Reply:
Good callout, let's chat about this in gather, when you get sometime!